packages/http-client-csharp/generator/Microsoft.Generator.CSharp/src/Configuration.cs

```diff
@@ -150,7 +150,7 @@ private static class Options
         /// <summary> Gets the namespace for the models. </summary>
         public string ModelNamespace { get; }
 
-        internal string OutputDirectory { get; }
+        public string OutputDirectory { get; }
```

Comment:
Looking at the usage, I don't think we need this to be public in order to accomplish the goal of adding an IsAzureArm property to the Azure plugin.

Reply:
Replied https://github.com/Azure/azure-sdk-for-net/pull/47374#discussion_r1865107185

`inputLibrary` is virtual, but without `Configuration.OutputDirectory` available it's impossible for sub-plugin to override it.
https://github.com/microsoft/typespec/blob/b9e5648e0bc106c47f66b9146562f2ba1341b62a/packages/http-client-csharp/generator/Microsoft.Generator.CSharp/src/CodeModelPlugin.cs#L67